console-subscriber/src/builder.rs

```diff
@@ -149,6 +158,283 @@ impl Builder {
+                });
+            })
+            .expect("console subscriber could not spawn thread");
+
+        layer
+    }
+}
+
+/// Initializes the console [tracing `Subscriber`][sub] and starts the console
+/// subscriber [`Server`] on its own background thread.
+///
+/// This function represents the easiest way to get started using
+/// tokio-console.
+///
+/// In addition to the [`TasksLayer`], which collects instrumentation data
+/// consumed by the console, the default [`Subscriber`][sub] initialized by this
+/// function also includes a [`tracing_subscriber::fmt`] layer, which logs
+/// tracing spans and events to stdout. Which spans and events are logged will
+/// be determined by the `RUST_LOG` environment variable.
+///
+/// **Note**: this function sets the [default `tracing` subscriber][default]
+/// for your application. If you need to add additional layers to a subscriber,
+/// see [`spawn`].
+///
+/// [default]: https://docs.rs/tracing/latest/tracing/dispatcher/index.html#setting-the-default-subscriber
+/// [sub]: https://docs.rs/tracing/latest/tracing/trait.Subscriber.html
+/// [`tracing_subscriber::fmt`]: https://docs.rs/tracing-subscriber/latest/tracing-subscriber/fmt/index.html
+/// [`Server`]: crate::Server
+///
+/// ## Configuration
+///
+/// Tokio console subscriber is configured with sensible defaults for most
+/// use cases. If you need to tune these parameters, several environmental
+/// configuration variables are available:
+///
+/// | **Environment Variable**            | **Purpose**                                                               | **Default Value** |
+/// |-------------------------------------|---------------------------------------------------------------------------|-------------------|
+/// | `TOKIO_CONSOLE_RETENTION`           | The number of seconds to accumulate completed tracing data                | 3600s (1h)        |
+/// | `TOKIO_CONSOLE_BIND`                | A HOST:PORT description, such as `localhost:1234`                         | `127.0.0.1:6669`  |
+/// | `TOKIO_CONSOLE_PUBLISH_INTERVAL`    | The number of milliseconds to wait between sending updates to the console | 1000ms (1s)       |
+/// | `TOKIO_CONSOLE_RECORD_PATH`         | The file path to save a recording                                         | None              |
+/// | `RUST_LOG`                          | Configures what events are logged events. See [`Targets`] for details.    | "error"           |
+///
+/// ## Further customization
+///
+/// To add additional layers or replace the format layer, replace
+/// `console_subscriber::init` with:
+///
+/// ```rust
+/// use tracing_subscriber::prelude::*;
+///
+/// let console_layer = console_subscriber::spawn();
+///
+/// tracing_subscriber::registry()
+///     .with(console_layer)
+///     .with(tracing_subscriber::fmt::layer())
+/// //  .with(..potential additional layer..)
+///     .init();
+/// ```
+///
```

Comment:
in general, i find that when there's a simple entry-point functions that just calls a couple other functions in the public API, it's often useful to explicitly document what that function does, so that users can easily reproduce the default configuration when switching to other APIs. so, we could maybe add something like this:
```suggestion
///
/// Calling `console_subscriber::init` is equivalent to the following:
/// ```rust
/// use console_subscriber::TasksLayer;
///
/// TasksLayer::builder().with_default_env().init();
/// ```
```